server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -431,10 +433,14 @@ public DynamicReference getDynamic(ColumnIdent ident, boolean forWrite) {
 
         switch (parentPolicy) {
             case DYNAMIC:
-                if (!forWrite) return null;
+                if (!forWrite) {
+                    if (!errorOnUnknownObjectKey) {
+                        throw new UnknownObjectKeyExceptionalControlFlow(ident.sqlFqn());
+                    }
+                    return null;
+                }
                 break;
             case STRICT:
-                if (forWrite) throw new ColumnUnknownException(ident.sqlFqn(), ident());
```

Comment:
seemed like returning null and throwing `ColumnUnknownException` have clear distinction

Reply:
reverted